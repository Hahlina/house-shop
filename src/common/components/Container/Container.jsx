import { Container as MUIContainer } from '@mui/material';

export const Container = ({ children }) => <MUIContainer maxWidth="lg">{children}</MUIContainer>;
