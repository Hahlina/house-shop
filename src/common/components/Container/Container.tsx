import type { FC, PropsWithChildren } from 'react';

import { Container as MUIContainer } from '@mui/material';

export const Container: FC<PropsWithChildren> = ({ children }) => (
    <MUIContainer maxWidth="lg" data-testid="container-id">
        {children}
    </MUIContainer>
);
