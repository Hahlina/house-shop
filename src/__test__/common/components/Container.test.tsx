import type { ReactNode } from 'react';
import { render, screen } from '@testing-library/react';

import { Container } from 'common/components/Container/Container';

describe('Container components', () => {
    let children: ReactNode;

    beforeEach(() => {
        children = <p>Children</p>;
    });

    it('should render children', () => {
        render(<Container>{children}</Container>);
        const textInContainer = screen.getByText(/children/i);
        expect(textInContainer).toBeInTheDocument();
    });

    it('should render without children', () => {
        render(<Container />);
        const containerElement = screen.queryByTestId('container-id');
        expect(containerElement).toBeInTheDocument();
    });

    it('should math snapshot without children', () => {
        render(<Container />);
        const containerElement = screen.queryByTestId('container-id');
        expect(containerElement).toMatchSnapshot();
    });

    it('should math snapshot with children', () => {
        render(<Container>{children}</Container>);
        const containerElement = screen.queryByTestId('container-id');
        expect(containerElement).toMatchSnapshot();
    });
});
