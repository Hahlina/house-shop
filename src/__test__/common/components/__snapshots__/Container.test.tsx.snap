// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container components should math snapshot with children 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  data-testid="container-id"
>
  <p>
    Children
  </p>
</div>
`;

exports[`Container components should math snapshot without children 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  data-testid="container-id"
/>
`;
